src/sentry/sentry_metrics/indexer/postgres.py

```diff
@@ -0,0 +1,78 @@
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Set
+
+from sentry.sentry_metrics.indexer.models import MetricsKeyIndexer
+from sentry.utils.services import Service
+
+
+class Record:
+    organization_id: int
+    string: str
+    id: int
+
+
+class PGStringIndexer(Service):  # type: ignore
+    """
+    Provides integer IDs for metric names, tag keys and tag values
+    and the corresponding reverse lookup.
+    """
+
+    __all__ = ("record", "resolve", "reverse_resolve", "bulk_record")
+
+    def _bulk_record(self, org_id: int, unmapped_strings: Set[str]) -> Any:
+        records = [
+            MetricsKeyIndexer(organization_id=org_id, string=string) for string in unmapped_strings
+        ]
+        return MetricsKeyIndexer.objects.bulk_create(records)
```

Comment:
If we hit unique constraint conflicts here what do we want to do? I assume you want this to be like a bulk `get_or_create`. I wrote something like this recently in `bulk_get_rule_status`.

tl;dr is that you can bulk create them with `ignore_conflicts=True`, but then you won't have the ids filled in. So you will then need to make a second query to fetch all the rows you just created.